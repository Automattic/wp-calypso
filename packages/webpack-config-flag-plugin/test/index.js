/**
 * External dependencies
 */
const fs = require( 'fs' );
const path = require( 'path' );
const rimraf = require( 'rimraf' );
const webpack = require( 'webpack' );
const ConfigFlagPlugin = require( '..' );

const fixturesDirectory = path.join( __dirname, 'fixtures' );
const buildDirectory = path.join( __dirname, 'build' );

describe( 'webpack-config-flag-plugin', () => {
	beforeAll( () => {
		rimraf.sync( buildDirectory );
	} );

	afterAll( () => {
		rimraf.sync( buildDirectory );
	} );

	test( 'should produce expected output', () =>
		new Promise( ( done ) => {
			const config = {
				context: fixturesDirectory,
				entry: path.join( fixturesDirectory, 'main.js' ),
				mode: 'production',
				devtool: false,
				optimization: {
					runtimeChunk: true,
					moduleIds: 'named',
					chunkIds: 'named',
					minimize: false,
				},
				output: {
					path: buildDirectory,
					pathinfo: false,
					globalObject: 'window',
				},
				resolve: {
					extensions: [ '.js' ],
					modules: [ fixturesDirectory ],
					alias: {
						'calypso/config': 'config',
					},
				},
				plugins: [ new ConfigFlagPlugin( { flags: { foo: true } } ) ],
			};

			webpack( config, ( err, stats ) => {
				expect( err ).toBeNull();
				expect( stats.compilation.errors ).toHaveLength( 0 );

				const outputFile = path.join( buildDirectory, 'main.js' );
				const outputFileContent = fs.readFileSync( outputFile, 'utf8' );
				expect( outputFileContent ).toMatchSnapshot( 'Output bundle should match snapshot' );

				done();
			} );
		} ) );
} );
