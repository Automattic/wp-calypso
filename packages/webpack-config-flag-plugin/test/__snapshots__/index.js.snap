// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-config-flag-plugin should produce expected output: Output bundle should match snapshot 1`] = `
"(window[\\"webpackChunk\\"] = window[\\"webpackChunk\\"] || []).push([[\\"main\\"],{

/***/ \\"./main.js\\":
/***/ (function(__unused_webpack_module, __unused_webpack___webpack_exports__, __webpack_require__) {

\\"use strict\\";

// NAMESPACE OBJECT: ./default-import/config/index.js
var default_import_config_namespaceObject = {};
__webpack_require__.r(default_import_config_namespaceObject);

// NAMESPACE OBJECT: ./named-import/config/index.js
var named_import_config_namespaceObject = {};
__webpack_require__.r(named_import_config_namespaceObject);

// NAMESPACE OBJECT: ./namespace-import/config/index.js
var namespace_import_config_namespaceObject = {};
__webpack_require__.r(namespace_import_config_namespaceObject);

;// CONCATENATED MODULE: ./config.js
function isEnabled( flag ) {
	return 'argh';
}

/* harmony default export */ var config = ({ isEnabled });

;// CONCATENATED MODULE: ./default-import/config/module.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./default-import/config/function.js


function fn() {
	// Should be replaced with true
	if ( true ) {
	}
}

;// CONCATENATED MODULE: ./default-import/config/wrong-flag.js


// Should NOT be replaced with true.
if ( config.isEnabled( 'bar' ) ) {
}

;// CONCATENATED MODULE: ./default-import/config/rename.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./default-import/config/wrong-path.js


// Should NOT be replaced with true
if ( default_import_config_namespaceObject.default.isEnabled( 'foo' ) ) {
}

;// CONCATENATED MODULE: ./default-import/config/shadowed-by-param.js


function shadowed_by_param_fn( config ) {
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./default-import/config/shadowed-by-var.js


function shadowed_by_var_fn() {
	const config = { isEnabled: () => false };
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./default-import/config/index.js








;// CONCATENATED MODULE: ./default-import/calypso-config/function.js


function function_fn() {
	// Should be replaced with true
	if ( true ) {
	}
}

;// CONCATENATED MODULE: ./default-import/calypso-config/module.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./default-import/calypso-config/rename.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./default-import/calypso-config/shadowed-by-param.js


function calypso_config_shadowed_by_param_fn( config ) {
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./default-import/calypso-config/shadowed-by-var.js


function calypso_config_shadowed_by_var_fn() {
	const config = { isEnabled: () => false };
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./default-import/calypso-config/wrong-flag.js


// Should NOT be replaced with true.
if ( config.isEnabled( 'bar' ) ) {
}

;// CONCATENATED MODULE: ./default-import/calypso-config/index.js







;// CONCATENATED MODULE: ./default-import/index.js



;// CONCATENATED MODULE: ./named-import/config/module.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./named-import/config/function.js


function config_function_fn() {
	// Should be replaced with true
	if ( true ) {
	}
}

;// CONCATENATED MODULE: ./named-import/config/wrong-flag.js


// Should NOT be replaced with true.
if ( isEnabled( 'bar' ) ) {
}

;// CONCATENATED MODULE: ./named-import/config/rename.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./named-import/config/wrong-path.js


// Should NOT be replaced with true
if ( (0,named_import_config_namespaceObject.isEnabled)( 'foo' ) ) {
}

;// CONCATENATED MODULE: ./named-import/config/shadowed-by-param.js


function config_shadowed_by_param_fn( isEnabled ) {
	// Should NOT be replaced with true
	if ( isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./named-import/config/shadowed-by-var.js


function config_shadowed_by_var_fn() {
	const isEnabled = () => false;
	// Should NOT be replaced with true
	if ( isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./named-import/config/index.js








;// CONCATENATED MODULE: ./named-import/calypso-config/function.js


function calypso_config_function_fn() {
	// Should be replaced with true
	if ( true ) {
	}
}

;// CONCATENATED MODULE: ./named-import/calypso-config/module.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./named-import/calypso-config/rename.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./named-import/calypso-config/shadowed-by-param.js


function named_import_calypso_config_shadowed_by_param_fn( isEnabled ) {
	// Should NOT be replaced with true
	if ( isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./named-import/calypso-config/shadowed-by-var.js


function named_import_calypso_config_shadowed_by_var_fn() {
	const isEnabled = () => false;
	// Should NOT be replaced with true
	if ( isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./named-import/calypso-config/wrong-flag.js


// Should NOT be replaced with true.
if ( isEnabled( 'bar' ) ) {
}

;// CONCATENATED MODULE: ./named-import/calypso-config/index.js







;// CONCATENATED MODULE: ./named-import/index.js



;// CONCATENATED MODULE: ./namespace-import/config/module.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./namespace-import/config/function.js


function namespace_import_config_function_fn() {
	// Should be replaced with true
	if ( true ) {
	}
}

;// CONCATENATED MODULE: ./namespace-import/config/wrong-flag.js


// Should NOT be replaced with true.
if ( isEnabled( 'bar' ) ) {
}

;// CONCATENATED MODULE: ./namespace-import/config/rename.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./namespace-import/config/wrong-path.js


// Should NOT be replaced with true
if ( namespace_import_config_namespaceObject.isEnabled( 'foo' ) ) {
}

;// CONCATENATED MODULE: ./namespace-import/config/shadowed-by-param.js


function namespace_import_config_shadowed_by_param_fn( config ) {
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./namespace-import/config/shadowed-by-var.js


function namespace_import_config_shadowed_by_var_fn() {
	const config = { isEnabled: () => false };
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./namespace-import/config/index.js








;// CONCATENATED MODULE: ./namespace-import/calypso-config/function.js


function namespace_import_calypso_config_function_fn() {
	// Should be replaced with true
	if ( true ) {
	}
}

;// CONCATENATED MODULE: ./namespace-import/calypso-config/module.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./namespace-import/calypso-config/rename.js


// Should be replaced with true
if ( true ) {
}

;// CONCATENATED MODULE: ./namespace-import/calypso-config/shadowed-by-param.js


function namespace_import_calypso_config_shadowed_by_param_fn( config ) {
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./namespace-import/calypso-config/shadowed-by-var.js


function namespace_import_calypso_config_shadowed_by_var_fn() {
	const config = { isEnabled: () => false };
	// Should NOT be replaced with true
	if ( config.isEnabled( 'foo' ) ) {
	}
}

;// CONCATENATED MODULE: ./namespace-import/calypso-config/wrong-flag.js


// Should NOT be replaced with true.
if ( isEnabled( 'bar' ) ) {
}

;// CONCATENATED MODULE: ./namespace-import/calypso-config/index.js







;// CONCATENATED MODULE: ./namespace-import/index.js



;// CONCATENATED MODULE: ./main.js





/***/ })

},
0,[[\\"./main.js\\",\\"runtime~main\\"]]]);"
`;
